packages/graphiql-react/src/ui/tabs.tsx

```diff
@@ -1,67 +1,62 @@
-function TabCloseButton(props: { onClick: () => void }) {
+import { CloseIcon } from '../icons';
+import { compose } from '../utility/compose';
+import { UnStyledButton } from './button';
+
+import './tabs.css';
+
+type TabProps = {
+  isActive?: boolean;
+};
+
+export function Tab(props: TabProps & JSX.IntrinsicElements['div']) {
   return (
     <div
-      role="button"
-      aria-pressed={false}
-      className="close"
-      aria-label="Close Tab"
-      title="Close Tab"
-      onClick={ev => {
-        ev.stopPropagation();
-        props.onClick();
-      }}
-    />
+      {...props}
+      aria-selected={props.isActive}
+      className={compose(
+        'graphiql-tab',
+        props.isActive ? 'graphiql-tab-active' : '',
+        props.className,
+      )}>
+      {props.children}
+    </div>
   );
 }
 
-type TabProps = {
-  isActive: boolean;
-  title: string;
-  isCloseable: boolean;
-  onSelect: () => void;
-  onClose: () => void;
-  tabProps?: React.ButtonHTMLAttributes<{}>;
-};
-
-/**
- * Generic tab component that implements wai-aria tab spec
- */
-export function Tab(props: TabProps): React.ReactElement {
+function TabButton(props: JSX.IntrinsicElements['button']) {
   return (
-    <button
-      {...props.tabProps}
-      role="tab"
```

Comment:
oops! let's keep this. I don't know if this was on purpose, but anything that is a tab should have this role for the other aria props to work!
<img width="908" alt="Screenshot 2022-07-27 at 19 44 38" src="https://user-images.githubusercontent.com/1368727/181316988-7ff183f9-26a3-45b9-bfb6-5cf82c783bb3.png">

otherwise it appears everything is in order with the aria props, just this one is missing. it seems the tab components have been re-structured, but as you probably know, it should be used for the tab elements anywhere that tabpanel and tablist are used

Reply:
Very good catch! That should have moved to the `Tab` component, but seems like I messed that up.